internal/provider/resource_service_user_token.go

```diff
@@ -0,0 +1,209 @@
+package provider
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/planmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/stringplanmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+
+	"github.com/humanitec/humanitec-go-autogen"
+	"github.com/humanitec/humanitec-go-autogen/client"
+)
+
+// Ensure provider defined types fully satisfy framework interfaces
+var _ resource.Resource = &ResourceServiceUserToken{}
+
+func NewResourceServiceUserToken() resource.Resource {
+	return &ResourceServiceUserToken{}
+}
+
+// ResourceValue defines the resource implementation.
+type ResourceServiceUserToken struct {
+	client *humanitec.Client
+	orgId  string
+}
+
+// ValueModel describes the app data model.
+type ServiceUserTokenModel struct {
+	ID          types.String `tfsdk:"id"`
+	UserID      types.String `tfsdk:"user_id"`
+	Description types.String `tfsdk:"description"`
+	ExpiresAt   types.String `tfsdk:"expires_at"`
+	Token       types.String `tfsdk:"token"`
+}
+
+func (r *ResourceServiceUserToken) Metadata(ctx context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_service_user_token"
+}
+
+func (r *ResourceServiceUserToken) Schema(ctx context.Context, req resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		MarkdownDescription: "Tokens can be generated from service users and are used to interact with Humanitec’s API on the service user’s behalf.",
+
+		Attributes: map[string]schema.Attribute{
+			"id": schema.StringAttribute{
+				MarkdownDescription: "Identifier of the token. Must be unique for the user.",
+				Required:            true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
+			},
+			"user_id": schema.StringAttribute{
+				MarkdownDescription: "The service user ID.",
+				Required:            true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
+			},
+			"description": schema.StringAttribute{
+				MarkdownDescription: "A description of the token.",
+				Optional:            true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
+			},
+			"expires_at": schema.StringAttribute{
+				MarkdownDescription: "The time the token expires. If not set, the token will not expire.",
+				Optional:            true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
+			},
+			"token": schema.StringAttribute{
+				MarkdownDescription: "Unique token granting access to specific services within the platform.",
+				Computed:            true,
```

Comment:
This needs a `Sensitive:           true,` so the token isn't displayed in the output.

Reply:
Fixed!